tests/system/src/test/java/test/robot/javafx/scene/RobotTest.java

```diff
@@ -0,0 +1,342 @@
+        CountDownLatch onScrollLatch = new CountDownLatch(1);
+        CountDownLatch setSceneLatch = new CountDownLatch(1);
+        Button button = new Button("Scroll me");
+        InvalidationListener invalidationListener = observable -> setSceneLatch.countDown();
+        Platform.runLater(() -> {
+            button.setOnScroll(event -> {
+                button.setText("Scrolled");
+                onScrollLatch.countDown();
+            });
+            scene = new Scene(new HBox(button), SIZE, SIZE);
+            stage.sceneProperty().addListener(observable -> {
+                setSceneLatch.countDown();
+                stage.sceneProperty().removeListener(invalidationListener);
+            });
+            stage.setScene(scene);
+        });
+        waitForLatch(setSceneLatch, 5, "Timeout while waiting for scene to be set on stage.");
+        Platform.runLater(() -> {
+            int mouseX = (int) (scene.getWindow().getX() + scene.getX() +
+                    button.getLayoutX() + button.getLayoutBounds().getWidth() / 2);
+            int mouseY = (int) (scene.getWindow().getY() + scene.getY() +
+                    button.getLayoutY() + button.getLayoutBounds().getHeight() / 2);
+            robot.mouseMove(mouseX, mouseY);
+            robot.mouseWheel(5);
+        });
+        waitForLatch(onScrollLatch, 5, "Timeout while waiting for button.onScroll().");
+        Assert.assertEquals("mouse wheel should be scrolled 5 vertical units by Robot", "Scrolled", button.getText());
+    }
+
+    @Test
+    public void testPixelCapture() throws Exception {
+        CountDownLatch pixelCaptureLatch = new CountDownLatch(1);
+        CountDownLatch setSceneLatch = new CountDownLatch(1);
+        Pane pane = new StackPane();
+        InvalidationListener invalidationListener = observable -> setSceneLatch.countDown();
+        Platform.runLater(() -> {
+            pane.setBackground(new Background(new BackgroundFill(Color.CORNFLOWERBLUE, null, null)));
+            scene = new Scene(pane, SIZE, SIZE);
+            stage.sceneProperty().addListener(observable -> {
+                setSceneLatch.countDown();
+                stage.sceneProperty().removeListener(invalidationListener);
+            });
+            stage.setScene(scene);
+        });
+        waitForLatch(setSceneLatch, 5, "Timeout while waiting for scene to be set on stage.");
+        AtomicReference<Color> captureColor = new AtomicReference<>();
+        Thread.sleep(1000);
+        Platform.runLater(() -> {
+            int x = (int) stage.getX();
+            int y = (int) stage.getY();
+            captureColor.set(robot.getPixelColor(x + SIZE / 2, y + SIZE / 2));
+            pixelCaptureLatch.countDown();
+        });
+        waitForLatch(pixelCaptureLatch, 5, "Timeout while waiting for robot.getPixelColor().");
+        assertColorEquals(Color.CORNFLOWERBLUE, captureColor.get(), TOLERANCE);
+    }
+
+    @Test
+    public void testScreenCapture() throws Exception {
```

Comment:
Relating to the follow-on issue about adding new tests: we should have another test for screen capture using an asymmetrical image, so that it would fail if the image were right-left or top-bottom reversed.